llvm-bazel/terminfo.bzl

```
@@ -0,0 +1,217 @@
+#            allow_single_file = True,
+#        ),
+#        "external_terminfo": attr.label(
+#            doc = "The dependency that should be used for the external terminfo library.",
+#            mandatory = True,
+#        ),
+#    },
+#)
+
+def _find_c_compiler(repository_ctx):
+    """Returns the path to a plausible C compiler.
+
+    This routine will only reliably work on roughly POSIX-y systems as it
+    ultimately falls back on the `cc` binary. Fortunately, the thing we are
+    trying to use it for (detecting if a trivial source file can compile and
+    link against a particular library) requires very little.
+    """
+    cc = repository_ctx.which(repository_ctx.os.environ.get("CC"))
+    if not cc:
+        cc = repository_ctx.which("clang")
+    if not cc:
+        cc = repository_ctx.which("gcc")
+    if not cc:
+        cc = repository_ctx.which("cc")
+    return cc
+
+def _try_link(repository_ctx, cc, source, linker_flags):
+    """Returns `True` if able to link the source with the linker flag.
+
+    Given a source file that contains references to library routines, this
+    will check that when linked with the provided linker flag, those
+    references are successfully resolved. This routine assumes a generally
+    POSIX-y and GCC-ish compiler and environment and shouldn't be expected to
+    work outside of that.
+    """
+    cmd = [
+      cc,
+      # Force discard the linked executable.
+      "-o", "/dev/null",
+      # Leave language detection to the compiler.
+      source,
+    ]
+    # The linker flag must be valid for a compiler invocation of the link step,
+    # so just append them to the command.
+    cmd += linker_flags
+    exec_result = repository_ctx.execute(cmd, timeout = 20)
+    return exec_result.return_code == 0
+
+def _llvm_terminfo_system_impl(repository_ctx):
+    # LLVM doesn't need terminfo support on Windows, so just disable it.
+    if repository_ctx.os.name.lower().find("windows") != -1:
+        _llvm_terminfo_disable_impl(repository_ctx)
+        return
+
+    if len(repository_ctx.attr.system_linkopts) > 0:
+        linkopts = repository_ctx.attr.system_linkopts
+    else:
+        required = repository_ctx.attr.system_required
+
+        # Find a C compiler we can use to detect viable linkopts on this system.
+        cc = _find_c_compiler(repository_ctx)
+        if not cc:
+            if required:
+                fail("Failed  to find a C compiler executable")
+            else:
+                _llvm_terminfo_disable_impl(repository_ctx)
+                return
+
+        # Get the source file we use to detect successful linking of terminfo.
+        source = repository_ctx.path(repository_ctx.attr._terminfo_test_source)
+
+        # Collect the candidate linkopts and wrap them into a list. Ideally,
+        # these would be provided as lists, but Bazel doesn't currently
+        # support that. See feature request #12178.
```

Can you link to the Bazel feature request. Otherwise this implies an issue in this repository.

```suggestion
        # support that. See https://github.com/bazelbuild/bazel/issues/12178
```